### PR TITLE
feat: add spinner in button

### PR DESCRIPTION
**Pull request by jmcollin**
### ❓ Types of changes

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality)
- [ ] 📦 New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes
- [ ] The component exists on old Prestashop UIKit and my pull request on [migrating documentation](https://github.com/PrestaShopCorp/devdocs.uikit.prestashop.com) is accepted.

### CODE REVIEW

**Review by GuillaumeGuerin1 — APPROVED** [review]

File: packages/components/button/stories/button.stories.ts [hunk 1]
```diff
@@ -375,6 +384,32 @@ Fluid.parameters = {
   },
 }
 
+export const WithSpinner: Story = ButtonTemplate.bind({})
```
Comment (Kelhel) [inline packages/components/button/stories/button.stories.ts:387]: On peut la renomer `Loading` plutôt ?

File: packages/components/button/stories/button.stories.ts [hunk 2]
```diff
@@ -375,6 +384,32 @@ Fluid.parameters = {
...
+      <!--VueJS Snippet -->
+      <puik-button left-icon="shopping_cart" loading>My button</puik-button>
+
+      <!--HTML/CSS Snippet-->
+      <button class="puik-button puik-button--primary">
+        <div class="puik-spinner-loader puik-spinner-loader--md puik-spinner-loader--reverse puik-button__loading" aria-live="polite" role="status">
+          <div class="puik-spinner-loader__spinner" aria-hidden="true"></div>
+        </div>
+        My button
+      </button>
```
Comment (Kelhel) [inline packages/components/button/stories/button.stories.ts:406]: Tu peux ne pas indenter les parties template de la doc stp ? ça évite les bugs d'affichage

File: packages/components/button/stories/button.stories.ts [hunk 3]
```diff
@@ -43,6 +43,14 @@ export default {
         },
       },
     },
+    loading: {
+      description: 'Set the spinner loader inside the button',
+      table: {
+        defaultValue: {
+          summary: false,
+        },
```
Comment (Kelhel) [inline packages/components/button/stories/button.stories.ts:51]:
```suggestion
        defaultValue: {
          summary: false,
        },
        type: {
          summary: 'boolean',
        },
```

File: packages/components/button/stories/button.stories.ts [hunk 4]
```diff
@@ -43,6 +43,14 @@ export default {
         },
       },
     },
+    loading: {
+      description: 'Set the spinner loader inside the button',
```
Comment (Kelhel) [inline packages/components/button/stories/button.stories.ts:47]:
```suggestion
      description: 'Display spinner loader inside button, set it disabled',
```

**Review by Kelhel — CHANGES_REQUESTED** [review]
Il faut ajouter l'import des fichier de styles liés au spinner loader dans `./packages/components/button/style/css.ts` et `./packages/components/button/style/index.ts`

**Comment by Kelhel** [comment]
Peux-tu aussi ajouter un test sur la prop loading stp ?

File: packages/components/button/stories/button.stories.ts (hunk 1 above)
Comment (jmcollin) [reply packages/components/button/stories/button.stories.ts:390]: `WithLoading` ou juste `Loading` ?

**Review by TakeshiDaveau — APPROVED** [review]

File: packages/components/button/stories/button.stories.ts (hunk 1 above)
Comment (jmcollin) [reply packages/components/button/stories/button.stories.ts]: Vu ensemble 

Merci ;)

File: packages/components/button/stories/button.stories.ts (hunk 2 above)
Comment (jmcollin) [reply packages/components/button/stories/button.stories.ts]: Vu ensemble 

Merci ;)

File: packages/components/button/src/button.vue [hunk 5]
```diff
@@ -6,18 +6,26 @@
...
+
+    <puik-spinner-loader
+      v-if="loading"
+      :size="size"
+      color="reverse"
+      class="puik-button__loading"
+    />
+
     <slot></slot>
     <puik-icon
```
Comment (Kelhel) [inline packages/components/button/src/button.vue:30]: So the right icon stays if the loading is enabled ?

Comment (jmcollin) [reply packages/components/button/src/button.vue:30]: Maybe I don't have this info

Comment (Kelhel) [reply packages/components/button/src/button.vue:30]: OK ... Also the contrast with different variant doesn't seems enough, the loader is invisible on variant `text`

**Comment by mattgoud** [comment]
codebase outdated - see #332

**Pull request closed without merging**